docx/from/docxtotei.xsl

```diff
@@ -1,713 +1,743 @@
+   </xsl:template>
+
+
+   <doc xmlns="http://www.oxygenxml.com/ns/doc/xsl">
+     <desc>Process the text by high-level divisions
+     </desc>
+   </doc>
+   <xsl:template name="mainProcess">
+     <xsl:param name="extrarow"  tunnel="yes"/>
+     <xsl:param name="extracolumn"   tunnel="yes"/>
+     <!-- 
+     group all paragraphs that form a first level section.
+      -->
+      <xsl:for-each-group select="w:sdt|w:p|w:tbl"
+        group-starting-with="w:p[tei:isFirstlevel-heading(.)]">
+
+        <xsl:choose>
+
+          <!-- We are dealing with a first level section, we now have
+         to further divide the section into subsections that we can then
+         finally work on -->
+
+    <xsl:when test="tei:is-heading(.)">
+      <xsl:call-template name="group-by-section"/>
+    </xsl:when>
+
+    <xsl:when test="tei:is-front(.)">
+      <front>
+        <xsl:apply-templates select="." mode="inSectionGroup"/>
+      </front>
+    </xsl:when>
+
+    <!-- We have found some loose paragraphs. These are most probably
+         front matter paragraps. We can simply convert them without further
+         trying to split them up into sub sections. -->
+    <xsl:otherwise>
+      <xsl:apply-templates select="." mode="inSectionGroup"/>
+    </xsl:otherwise>
+  </xsl:choose>
+</xsl:for-each-group>
+
+<!-- I have no idea why I need this, but I apparently do. 
+     //TODO: find out what is going on-->
+      <xsl:apply-templates select="w:sectPr" mode="paragraph"/>
+    </xsl:template>
+    <doc xmlns="http://www.oxygenxml.com/ns/doc/xsl">
+      <desc>
+        <p>Bookmarks in section mode</p>
+        <p>
+          There are certain elements that we don't really care about, but that
+          force us to regroup everything from the next sibling on.
+
+          @see grouping in construction of headline outline.
+        </p>
+      </desc>
+    </doc>
+    <xsl:template match="w:bookmarkStart|w:bookmarkEnd"
+      mode="inSectionGroup">
+      <xsl:for-each-group select="current-group() except ." group-adjacent="1">
+        <xsl:apply-templates select="." mode="inSectionGroup"/>
+      </xsl:for-each-group>
+    </xsl:template>
+
+
+    <doc xmlns="http://www.oxygenxml.com/ns/doc/xsl">
+      <desc>
+        <p>Bookmarks in normal mode</p>
+        <p>Copy bookmarks for processing in pass 2</p>
+      </desc>
+    </doc>
+    <xsl:template match="w:bookmarkStart|w:bookmarkEnd" >
+      <xsl:choose>
+        <xsl:when test="@w:name='_GoBack'"/>
+        <xsl:when test="starts-with(@w:name,'_')">
+          <xsl:copy-of select="."/>
+        </xsl:when>
+        <xsl:when test="self::w:bookmarkEnd"/>
+        <xsl:otherwise>
+          <ANCHOR>
+            <xsl:attribute name="xml:id" select="@w:name"/>
+          </ANCHOR>
+        </xsl:otherwise>
+      </xsl:choose>
+    </xsl:template>
+
+    <xsl:template name="removeNestedTables" match="w:txbxContent/w:tbl" mode="passOneHalf">
+      Could not process table here.  Table was removed.
```

Comment:
So I can wrap the error in a paragraph.  Because of the step, the wrapping has to be in Word XML format.  I was able to reverse engineer the paragraph though.